tools/bin/ecflow/processes/main/meteorological/get_BC_metdata.py

```diff
@@ -0,0 +1,143 @@
+    model_tools.replace_var_pythonic_config(old_config_file, new_config_file,
+                                            header=None, **kwargs)
+    # an abbreviation and a full name is needed
+    varnames = [('pr', 'precipitation_amount'), ('tmmn', 'air_temperature'),
+                ('tmmx', 'air_temperature'), ('vs', 'wind_speed'),
+                ('srad', 'surface_downwelling_shortwave_flux_in_air'),
+                ('sph', 'specific_humidity')]
+    # set up a connection to the NKN network
+    ssh = paramiko.SSHClient()
+    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+    ssh.load_host_keys(os.path.expanduser(
+        os.path.join("~", ".ssh", "known_hosts")))
+    ssh.connect('reacchdb.nkn.uidaho.edu', username='vicmet', password='cl!m@te')
+    sftp = ssh.open_sftp()
+    # access data files
+    for year in range(start_date.year, end_date.year + 1):
+        # transfer the meteorological files
+        for var in varnames:
+            source_file = os.path.join(source_loc, '{0}_BC_{1}.nc'.format(
+                var[0], year))
+            local_file = os.path.join(dest_loc, '{0}_{1}.nc'.format(
+                var[0], year))
+            sftp.get(source_file, local_file)
+    sftp.close()
+    ssh.close()
+    met_dsets = {}
+    if start_date.year != end_date.year:
+        for var in varnames:
+            ds_lastyear = xr.open_dataset(os.path.join(
+                dest_loc, '{0}_{1}.nc'.format(var[0], start_date.year)))
+            ds_thisyear = xr.open_dataset(os.path.join(
+                dest_loc, '{0}_{1}.nc'.format(var[0], end_date.year)))
+            # concatenate the two datasets and add general attributes
+            xds = xr.concat([ds_lastyear, ds_thisyear], 'day')
+
+            # place data in dict, the variable abbreviation is used as key
+            met_dsets[var[0]] = xds
+    else:
+        for var in varnames:
+            met_dsets[var[0]] = xr.open_dataset(os.path.join(
+                dest_loc, '{0}_{1}.nc'.format(var[0], start_date.year)))
+    for var in ('tmmn', 'tmmx'):
+        # Perform units conversion
+        units_in.convert(met_dsets[var].air_temperature.values[:], units_out,
+                         inplace=True)
+        # Fix _FillValue after unit conversion
+        met_dsets[var].air_temperature.values[
+            met_dsets[var].air_temperature < -30000] = -32767.
+        # Change variable names so that tmmn and tmax are different
+        met_dsets[var].rename({'air_temperature': var}, inplace=True)
+    merge_ds = xr.merge(list(met_dsets.values()))
+    # clip to relevant dates
+    merge_ds = merge_ds.loc[dict(day=slice(start_date_format,
+                                           end_date_format))]
+    merge_ds.transpose('day', 'lat', 'lon')
+    # MetSim requires time dimension be named "time"
+    merge_ds.rename({'day': 'time'}, inplace=True)
+    # Make sure tmax >= tmin always
+    tmin = np.copy(merge_ds['tmmn'].values)
```

Comment:
do you want to give an error if this isn't the case? Or a warning? Maybe a warning would be useful

Reply:
Added a warning at commit f2ca170. It is not an error in this script. The method used for downscaling the data does not require tmin to be less than tmax; however, MetSim and VIC do.

Reply:
For future reference, I've use the following approach in the past: 

```Python
    TINY_TEMP = 1e-10
    ...
    ds['t_max'] = ds['t_max'].where(ds['t_max'] > ds['t_min'],
                                    ds['t_min'] + TINY_TEMP)
```

This does everything the swap_values logic does before but sticks with numpy instead of falling back to numpy.